src/argilla/client/feedback/metrics/agreement_metrics.py

```diff
@@ -0,0 +1,273 @@
+    LabelQuestion: binary_distance,
+    MultiLabelQuestion: edit_distance,
+    RatingQuestion: interval_distance,
+    RankingQuestion: edit_distance,
+}
+
+
+class AgreementMetric:
+    def __init__(self, dataset: "FeedbackDataset", question_name: str) -> None:
+        """Initializes a `AgreementMetric` object to compute agreement metrics on
+        a `FeedbackDataset` for a given question.
+
+        Args:
+            dataset: FeedbackDataset to compute the metrics.
+            question_name: Name of the question for which we want to analyse the agreement.
+
+        Raises:
+            NotImplementedError: If the question type is not supported.
+        """
+        self._dataset = dataset
+        self._question_name = question_name
+        self._question_type = type(self._dataset.question_by_name(question_name))
+        if self._question_type == TextQuestion:
+            raise NotImplementedError(f"No metrics are defined currently for {self._question_type.__name__}")
+        self._allowed_metrics = METRICS_PER_QUESTION[self._question_type]
+
+    def compute(self, metric_names: Union[str, List[str]], **kwargs) -> List[AgreementMetricResult]:
+        """Computes the agreement metrics for the given question.
+
+        Args:
+            metric_names: name or list of names for the metrics to compute. i.e. `alpha`.
+            kwargs: additional arguments to pass to the metric.
+
+        Raises:
+            ValueError: If the metric name is not supported for the given question.
+
+        Returns:
+            agreement_metrics: A list of `AgreementMetricResult` objects for the dataset.
+        """
+        if isinstance(metric_names, str):
+            metric_names = [metric_names]
+
+        if any([metric not in self._allowed_metrics for metric in metric_names]):
+            raise ValueError(
+                f"Metrics allowed for question {self._question_name}: {list(self._allowed_metrics.keys())}"
+            )
+
+        metric_classes = [(metric_name, self._allowed_metrics[metric_name]) for metric_name in metric_names]
+
+        dataset = prepare_dataset_for_annotation_task(self._dataset, self._question_name)
+
+        distance_function = kwargs.get("distance_function")
+        if not distance_function:
+            distance_function = QUESTION_TO_DISTANCE[self._question_type]
+
+        metrics = []
+        for metric_name, metric_cls in metric_classes:
+            metric = metric_cls(annotated_dataset=dataset, distance_function=distance_function)
+            result = metric.compute(**kwargs)
+            metrics.append(AgreementMetricResult(metric_name=metric_name, result=result))
+
+        return metrics
+
+
+class AnnotationTaskMetricBase(ABC):
+    """Base class for Agreement metrics."""
+
+    def __init__(self, annotated_dataset=None, distance_function: Callable = None) -> None:
+        """
+        Args:
+            annotated_dataset: Annotated dataset as expected by the metric.
```

Comment:
maybe I miss type hints here? also the distance function might be set as `lambda: x` by default.